### DISCUSSION TITLE
State value override

### DISCUSSION THREAD

**Discussion by paulkelele** [OpenIDC/mod_auth_openidc#701] [General]
hi,

Is it possible to change the value of the state parameter, to put an arbitrary value there instead of having a value generated by the mod_auth_openidc module ?

I looked at the documentation here: https://github.com/zmartzone/mod_auth_openidc/blob/v2.4.5/auth_openidc.conf

but I did not find anything convincing.

 Thank you

**Comment by zandbelt** [comment]
no, that is not possible but depending on what you're trying to achieve, there may be another way of doing that

**Reply by paulkelele** [reply]
well it would be a question of differentiating the origins of the requests according to the client of the application which makes his request by arbitrarily assigning a state to each client.
For example client1 state=00000001 and so on ...
Or something close to that by creating subdomains for each customer with this parameter discriminating by state value for each customer. it's the idea.

**Comment by zandbelt** [comment]
you should be able to do that with an application specific parameter that you'd add to the URL that is being accessed, without bothering with the authentication request/response at the module/OIDC level

**Comment by paulkelele** [comment]
Okay. I understand. Thanks Hans
